### ISSUE TITLE
Using pyproject.toml triggers special pip behavior

### ISSUE THREAD

**Issue by exarkun** [hawkowl/towncrier#123]
I shouldn't have to kick pip into a new, incompatible mode just to configure towncrier.  towncrier should provide an alternate configuration mechanism that avoids pep 518 and pyproject.toml completely.

**Comment by exarkun** [comment]
One possible work-around would be to add an option to specify a file to use instead of pyproject.toml.  If an alternately named file is used it won't trigger the special processing from pep 518.

**Comment by zgoda-mobica** [comment]
This becomes even more problematic with pip 19:

```
editable mode is not supported for pyproject.toml-style projects. pip is processing this project as pyproject.toml-style because it has a pyproject.toml file. Since the project has a setup.py and the pyproject.toml has no "build-backend" key for the "build_system" value, you may pass --no-use-pep517 to opt out of pyproject.toml-style processing. See PEP 517 for details on pyproject.toml-style projects.
```

This requires special pip switches for editable installs.

**Comment by mauritsvanrees** [comment]
I don't know what the original issue text means, but the [comment from @zgoda-mobica](https://github.com/hawkowl/towncrier/issues/123#issuecomment-487528435) made me look up that message from pip.
Apparently more people have complained about this pip behavior, see [this thread](https://discuss.python.org/t/pip-19-1-and-installing-in-editable-mode-with-pyproject-toml/1553). (I haven't fully read it.)
And a couple of hours ago, a [pip PR](https://github.com/pypa/pip/pull/6449) was merged that reverts the changes that caused the editable mode problem. So should be fixed with a new pip release.

@exarkun Is this the problem you meant or was it something else?

**Comment by exarkun** [comment]
Note the issue was filed more than 4 days ago so I don't think the recently merged pip PR solves the issue.

I'll try to recreate the problem and post more details.

**Comment by exarkun** [comment]
Actually, maybe I won't, because #127 was merged and provides a work-around.  Not sure why this issue is still open at all.

**Comment by exarkun** [comment]
I'm not sure how to reproduce the original problem now that there's a new pip release with a new behavior, anyway.  The problem was probably originally encountered via tox which wants to do extra management of installed versions.  The obvious attempt at pinning an older version of pip doesn't produce the problem.  Consider #127 is merged I'm not sure it's worth digging through the history further to try to understand.  If it's still broken, I'm sure I'll be back. ;)

I'm gonna call this fixed by #127.
